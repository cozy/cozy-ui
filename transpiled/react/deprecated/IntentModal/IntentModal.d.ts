export default IntentModal;
/**
 * Render a modal for the specified intent.
 *
 * The modal for an intent takes the majority of the viewport.
 *
 * @deprecated Please use IntentIframe with Dialog instead
 */
declare class IntentModal extends React.Component<any, any, any> {
    constructor(props: any, context: any);
    state: {
        closable: boolean;
    };
    hideCross: () => void;
    showCross: () => void;
    dismiss: (evt: any) => void;
    render(): JSX.Element;
}
declare namespace IntentModal {
    const propTypes: any;
    const defaultProps: any;
}
import React from "react";
