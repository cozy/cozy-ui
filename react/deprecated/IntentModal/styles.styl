@require 'settings/breakpoints'
@require 'tools/mixins'

.intentModal
    height 90vh

    // takes 100% of the screen on mobile
    +small-screen()
        height 100%
        width 100%

    // child takes all height
    & > div
        height 100%

        // cozy-collect resizes this element with a max-width of 931px
        // they will remove it as it is a bug on their part. Then
        // we can remove this code
        max-width unset !important

    .intentModal__cross
        top rem(4)
        right rem(8)
