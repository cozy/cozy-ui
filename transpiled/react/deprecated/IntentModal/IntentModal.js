import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

import once from 'lodash/once';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
var styles = {
  "intentModal": "styles__intentModal___1NbfY",
  "intentModal__cross": "styles__intentModal__cross___1ISTs"
};
import IntentIframe from "cozy-ui/transpiled/react/IntentIframe";
import createDepreciationLogger from "cozy-ui/transpiled/react/helpers/createDepreciationLogger";
import Modal from "cozy-ui/transpiled/react/deprecated/Modal";
var logIntentModalDepecrated = createDepreciationLogger();
/**
 * Render a modal for the specified intent.
 *
 * The modal for an intent takes the majority of the viewport.
 *
 * @deprecated Please use IntentIframe with Dialog instead
 */

var IntentModal = /*#__PURE__*/function (_Component) {
  _inherits(IntentModal, _Component);

  var _super = _createSuper(IntentModal);

  function IntentModal(props, context) {
    var _this;

    _classCallCheck(this, IntentModal);

    _this = _super.call(this, props, context);

    _defineProperty(_assertThisInitialized(_this), "hideCross", function () {
      _this.setState({
        closable: false
      });
    });

    _defineProperty(_assertThisInitialized(_this), "showCross", function () {
      _this.setState({
        closable: true
      });
    });

    _defineProperty(_assertThisInitialized(_this), "dismiss", once(function (evt) {
      var dismissAction = _this.props.dismissAction;
      dismissAction && dismissAction(evt);
    }));

    logIntentModalDepecrated('The IntentModal component has been deprecated and should be replaced by IntentIframe wrapped in Dialog');
    _this.state = {
      closable: true
    };
    return _this;
  }

  _createClass(IntentModal, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          action = _this$props.action,
          doctype = _this$props.doctype,
          options = _this$props.options,
          onComplete = _this$props.onComplete,
          onError = _this$props.onError,
          create = _this$props.create;
      var modalProps = Object.keys(Modal.propTypes).reduce(function (props, key) {
        return _objectSpread(_objectSpread({}, props), {}, _defineProperty({}, key, _this2.props[key]));
      }, {});
      return /*#__PURE__*/React.createElement(Modal, _extends({}, modalProps, {
        closable: this.state.closable,
        key: "modal",
        className: styles.intentModal,
        closeBtnClassName: styles.intentModal__cross,
        dismissAction: this.dismiss,
        overflowHidden: true
      }), /*#__PURE__*/React.createElement(IntentIframe, {
        action: action,
        create: create,
        data: options,
        onCancel: this.dismiss,
        onError: onError,
        onTerminate: onComplete,
        type: doctype,
        onHideCross: this.hideCross,
        onShowCross: this.showCross
      }));
    }
  }]);

  return IntentModal;
}(Component);

IntentModal.propTypes = _objectSpread(_objectSpread({}, Modal.propTypes), {}, {
  /** What should happen when the intent has completed */
  onComplete: PropTypes.func.isRequired,

  /** What should happen if an intent error occurs */
  onError: PropTypes.func,

  /** Action you want to execute */
  action: PropTypes.string.isRequired,

  /** Doctype on which you want to execute the action */
  doctype: PropTypes.string.isRequired
});
IntentModal.defaultProps = _objectSpread({}, Modal.defaultProps);
export default IntentModal;