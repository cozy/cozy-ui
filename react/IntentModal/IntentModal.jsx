import React, { Component } from 'react'
import styles from './styles.styl'
import PropTypes from 'prop-types'

import once from 'lodash/once'

import IntentIframe from '../IntentIframe'
import Modal from '../Modal'

/**
 * Render a modal for the specified intent.
 *
 * The modal for an intent takes the majority of the viewport.
 */
class IntentModal extends Component {

  // As dismissAction is passed twice to the modal, both for closing and for
  // intent cancellation, we need to ensure that it is only actually
  // called once.
  // FIXME: this should be fixed by diferenciating dismissAction (for closing
  // modal) and onCancel (for intent cancellation), but it implies deprecating
  // dismissAction first, ensure legacy, prevent regressions, etc.
  dismiss = once(() => {
    const { dismissAction } = this.props
    dismissAction && dismissAction()
  })

  render() {
    const {
      closable,
      overflowHidden,
      action,
      doctype,
      options,
      onComplete,
      onError,
      create,
      into
    } = this.props

    return (
      <Modal
        key="modal"
        closable={closable}
        overflowHidden={overflowHidden}
        className={styles.intentModal}
        crossClassName={styles.intentModal__cross}
        dismissAction={this.dismiss}
        into={into}
      >
        <IntentIframe
          action={action}
          create={create}
          data={options}
          onCancel={this.dismiss}
          onError={onError}
          onTerminate={onComplete}
          type={doctype}
        />
      </Modal>
    )
  }
}

IntentModal.propTypes = {
  /** What should happen when the intent has completed */
  onComplete: PropTypes.func.isRequired,
  /** What should happen if an intent error occurs */
  onError: PropTypes.func,
  /** What to do to dismiss the modal */
  dismissAction: PropTypes.func.isRequired,
  /** Action you want to execute */
  action: PropTypes.string.isRequired,
  /** Doctype on which you want to execute the action */
  doctype: PropTypes.string.isRequired,
  /** Display the cross and enable click outside and escape key to close */
  closable: PropTypes.bool,
  /** Where the modal should be rendered in the DOM */
  into: PropTypes.string
}

IntentModal.defaultProps = {
  closable: true
}

export default IntentModal
