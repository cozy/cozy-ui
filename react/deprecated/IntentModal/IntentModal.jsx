import once from 'lodash/once'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import styles from './styles.styl'
import IntentIframe from '../../IntentIframe'
import createDepreciationLogger from '../../helpers/createDepreciationLogger'
import Modal from '../Modal'

const logIntentModalDepecrated = createDepreciationLogger()

/**
 * Render a modal for the specified intent.
 *
 * The modal for an intent takes the majority of the viewport.
 *
 * @deprecated Please use IntentIframe with Dialog instead
 */
class IntentModal extends Component {
  constructor(props, context) {
    super(props, context)

    logIntentModalDepecrated(
      'The IntentModal component has been deprecated and should be replaced by IntentIframe wrapped in Dialog'
    )
    this.state = {
      closable: true
    }
  }

  hideCross = () => {
    this.setState({
      closable: false
    })
  }

  showCross = () => {
    this.setState({
      closable: true
    })
  }
  // As dismissAction is passed twice to the modal, both for closing and for
  // intent cancellation, we need to ensure that it is only actually
  // called once.
  // FIXME: this should be fixed by diferenciating dismissAction (for closing
  // modal) and onCancel (for intent cancellation), but it implies deprecating
  // dismissAction first, ensure legacy, prevent regressions, etc.
  dismiss = once(evt => {
    const { dismissAction } = this.props
    dismissAction && dismissAction(evt)
  })

  render() {
    const { action, doctype, options, onComplete, onError, create } = this.props
    const modalProps = Object.keys(Modal.propTypes).reduce((props, key) => {
      return { ...props, [key]: this.props[key] }
    }, {})
    return (
      <Modal
        {...modalProps}
        closable={this.state.closable}
        key="modal"
        className={styles.intentModal}
        closeBtnClassName={styles.intentModal__cross}
        dismissAction={this.dismiss}
        overflowHidden
      >
        <IntentIframe
          action={action}
          create={create}
          data={options}
          onCancel={this.dismiss}
          onError={onError}
          onTerminate={onComplete}
          type={doctype}
          onHideCross={this.hideCross}
          onShowCross={this.showCross}
        />
      </Modal>
    )
  }
}

IntentModal.propTypes = {
  ...Modal.propTypes,
  /** What should happen when the intent has completed */
  onComplete: PropTypes.func.isRequired,
  /** What should happen if an intent error occurs */
  onError: PropTypes.func,
  /** Action you want to execute */
  action: PropTypes.string.isRequired,
  /** Doctype on which you want to execute the action */
  doctype: PropTypes.string.isRequired
}

IntentModal.defaultProps = { ...Modal.defaultProps }

export default IntentModal
