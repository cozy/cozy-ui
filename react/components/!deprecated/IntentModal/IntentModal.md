The IntentOpener component is useful to start an new intent modal from a click on a button. But sometimes you want/have to handle the modal opening state on the application side so you need to just render an Intent inside a modal. The IntentModal component is clearly for that use case. This is also the same modal code used by the IntentOpener component.

```
import IntentModal from 'cozy-ui/transpiled/react/IntentModal';
initialState = { modalOpened: false};

<div>
  <button onClick={()=>setState({ modalOpened: !state.modalOpened })}>
    Toggle an IntentModal OPEN io.cozy.files
  </button>
  {state.modalOpened &&
    <IntentModal
      dismissAction={() => setState({ modalOpened: false })}
      onComplete={res => {
          setState({ modalOpened: false })
          alert('intent has completed ! ' + JSON.stringify(res))
        }
      }
      action='OPEN'
      doctype='io.cozy.files'
      // you would not pass create normally as it defaults to
      // cozy.client.intents.create
      create={utils.fakeIntentCreate}
    /> }
</div>
```
