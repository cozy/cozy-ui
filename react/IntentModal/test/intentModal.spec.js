'use strict'
/* eslint-env jest */

import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

import IntentModal from '../IntentModal'

configure({ adapter: new Adapter() })

describe('IntentModal component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it(`renders as expected`, () => {
    const component = shallow(
      <IntentModal
        action="ANY"
        closable
        create={jest.fn().mockName('createMock')}
        doctype="io.cozy.whatever"
        dismissAction={jest.fn().mockName('onDismissMock')}
        onComplete={jest.fn().mockName('onCompleteMock')}
        onDismiss={jest.fn().mockName('dismissActionMock')}
        onError={jest.fn().mockName('onErrorMock')}
        options={{ key: 'value' }}
        overflowHidden
      />
    ).getElement()

    expect(component).toMatchSnapshot()
  })
})
