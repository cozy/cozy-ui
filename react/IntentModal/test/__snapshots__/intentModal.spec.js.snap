// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntentModal component renders as expected 1`] = `
<Unknown
  className="intentModal"
  closable={true}
  crossClassName="intentModal__cross"
  dismissAction={[Function]}
  into={undefined}
  overflowHidden={true}
>
  <IntentIframe
    action="ANY"
    create={[MockFunction createMock]}
    data={
      Object {
        "key": "value",
      }
    }
    onCancel={[Function]}
    onError={[MockFunction onErrorMock]}
    onTerminate={[MockFunction onCompleteMock]}
    type="io.cozy.whatever"
  />
</Unknown>
`;
